libraries/GCS_MAVLink/GCS.cpp

```diff
@@ -20,32 +20,87 @@ void GCS::send_text(MAV_SEVERITY severity, const char *fmt, ...)
     va_end(arg_list);
 }
 
-#define FOR_EACH_ACTIVE_CHANNEL(methodcall)          \
-    do {                                             \
-        for (uint8_t i=0; i<num_gcs(); i++) {        \
-            if (!chan(i).initialised) {              \
-                continue;                            \
-            }                                        \
-            if (!(GCS_MAVLINK::active_channel_mask() & (1 << (chan(i).get_chan()-MAVLINK_COMM_0)))) { \
-                continue;                            \
-            }                                        \
-            chan(i).methodcall;                      \
-        }                                            \
-    } while (0)
+void GCS::send_to_active_channels(const mavlink_message_t &msg)
+{
+    for (uint8_t i=0; i<num_gcs(); i++) {
+        if (!chan(i).initialised) {
+            continue;
+        }
+        if (!(GCS_MAVLINK::active_channel_mask() & (1 << (chan(i).get_chan()-MAVLINK_COMM_0)))) {
+            continue;
+        }
+        chan(i).send_message(msg);
```

Comment:
Does this do the space checking for the message?

Reply:
Caller's responsibility.  I've added a check into this method.

I've also added another patch with a SITL sanity check for attempting to write too much into a uart.  Seems that might have picked something up.  May split that patch into a separate PR.

Reply:
Because this iterates all the channels how can the caller check in advance?